src/components/searchBar/index.tsx

```diff
@@ -1,65 +1,168 @@
+import { useToggleVisibility } from '@pivanov/use-toggle-visibility';
+import React, {
   useCallback,
+  useEffect,
+  useRef,
   useState,
 } from 'react';
 
 import { Icon } from '@components/icon';
-import { cn } from '@utils/helpers';
+import { ModalShowJson } from '@components/modals/modalShowJson';
+import { useStoreChain } from '@stores';
+import {
+  cn,
+  debounce,
+  findBlockByNumber,
+  findExtrinsicById,
+  getBlockNumberByHash,
+} from '@utils/helpers';
+
+import { Results } from './Results';
 
+import type { IMappedBlockExtrinsic } from '@custom-types/block';
 interface SearchBarProps {
   label: string;
   classNames?: string;
+  type: 'all' | 'block' | 'extrinsics';
   onSearch?: (query: string) => void;
 }
 
-export const SearchBar = (props: SearchBarProps) => {
-  const { label, classNames } = props;
+export const SearchBar = ({ label, classNames, type }: SearchBarProps) => {
+  const blocksData = useStoreChain?.use?.blocksData?.();
   const [searchInput, setSearchInput] = useState('');
+  const [isResultsVisible, setIsResultsVisible] = useState(false);
+  const [results, setResults] = useState<{
+    blockNumber: number | null;
+    extrinsics: IMappedBlockExtrinsic[] | null;
+  }>({
+    blockNumber: null,
+    extrinsics: null,
+  });
+
+  const refSelectedExtrinsic = useRef<IMappedBlockExtrinsic | undefined>();
+  const searchBarRef = useRef<HTMLDivElement>(null);
+
+  const [ShowJsonModal, toggleVisibility] = useToggleVisibility(ModalShowJson);
+
+  const debounceSearch = debounce((input) => {
+    setIsResultsVisible(true);
+
+    let searchValue = input as string;
+    const startsWith0x = searchValue.startsWith('0x');
+
+    if (startsWith0x) {
+      searchValue = getBlockNumberByHash(blocksData, searchValue);
+    } else {
+      searchValue = searchValue.replace(/,/g, '');
+    }
+
+    const block = findBlockByNumber(blocksData, searchValue);
+    const extrinsicMatch = /^([0-9]+)-[0-9]+$/.test(searchValue);
+
+    if (extrinsicMatch) {
+      const extrinsic = findExtrinsicById(blocksData, searchValue);
+
+      extrinsic
+      && setResults({
+        blockNumber: null,
+        extrinsics: [extrinsic],
+      });
+    } else {
+      setResults({
+        blockNumber: block?.header.number ?? null,
+        extrinsics: block?.body?.extrinsics ?? null,
+      });
+    }
+
+  }, 500);
 
-  const handleSearchInputChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
-    setSearchInput(event.target.value);
+  const handleOpenModal = useCallback((e: React.MouseEvent<HTMLDivElement>) => {
+    const extrinsicId = e.currentTarget.getAttribute('data-extrinsic-id');
+    if (results.extrinsics) {
+      refSelectedExtrinsic.current = results.extrinsics.find(extrinsic => extrinsic.id === extrinsicId);
+    }
+    toggleVisibility();
+  }, [results, toggleVisibility]);
+
+  const handleSearchInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(e.target.value);
```

Comment:
is it necessary to have separate function to call  both debounceSearch and setSearchInput?